src/PhpSpec/Formatter/ProgressFormatter.php

```diff
@@ -81,24 +81,19 @@ function ($count) use ($total) {
     }
 
     /**
-     * @param $percents
+     * @param array $counts
      * @return array
      */
-    private function getBarLengths($percents)
+    private function getBarLengths($counts)
     {
         $stats = $this->getStatisticsCollector();
-        $specProgress = ($stats->getTotalSpecs()+1)/$stats->getTotalSpecsCount();
-
-        $barLengths = array_map(
-            function ($percent) use ($specProgress) {
-                $length = $percent / 2;
-                $res = $length == 0 || $length > 1 ? floor($length * $specProgress) : 1;
+        $specProgress = ($stats->getTotalSpecsCount() == 0) ? 1 : ($stats->getTotalSpecs())/$stats->getTotalSpecsCount();
+        $targetWidth = ceil($this->getIO()->getBlockWidth() * $specProgress);
+        asort($counts);
 
-                return $res;
-            },
-            $percents
-        );
-        asort($barLengths);
+        $barLengths = array_map(function($count) use ($targetWidth, $counts) {
```

Comment:
I think the length of the inline function would make the return harder to pick out if combined